test/e2e/apps/disruption.go

```diff
@@ -215,6 +197,34 @@ var _ = SIGDescribe("DisruptionController", func() {
 			}
 		})
 	}
+
+	It("Allows updates to pdb's", func() {
+		By("Creating a pdb that targets all three pods in a test replica set")
+		createPDBMinAvailableOrDie(cs, ns, intstr.FromInt(3))
+		createReplicaSetOrDie(cs, ns, 3, false)
+
+		By("First trying to evict a pod which shouldn't be evictable")
+		pod, err := locateARunningPod(cs, ns)
+		Expect(err).NotTo(HaveOccurred())
+
+		e := &policy.Eviction{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: ns,
+			},
+		}
+		waitForPodsOrDie(cs, ns, 3) // make sure that they are running and so would be evictable with a different pdb
```

Comment:
what is the difference between waitForPodsOrDie and locateARUnningPod

Reply:
"Wait for pods" means let everything come to equilibrium. "Locate a running Pod" means choose a victim to try to evict.

Reply:
thanks, just looked at the code. Since we are evicting after waiting for all pods to be running, we can simply just waitForPodsOrDie and then evict because we just tested all are running. waitForPods could return one of the running pods to be evicted. Not a must do optimization , but prevents listing the pods twice.

Reply:
Having had a think about this I'd rather not combine these two operations (waiting for test setup equilibrium and selecting a pod for eviction) as I feel they are sufficiently different in intent and useful separately - although they do mean that the API is called twice.